Corrected incorrect markdown handling added syntax highlighting

eeyun opened the pull request:
This PR changes the markdown handling engine to redcarpet which has a more common syntax handling than kramdown.

Because of this, some of the new tutorial content had to be brought in line with redcarpet's syntax preferences.

It also adds in highlight.js for syntax highlighting over all of our codeblocks and no longer relies on middleman's syntax highlighter.

Signed-off-by: Ian Henry <ihenry@chef.io>

thesentinels commented:
Thanks for the pull request! Here is what will happen next:
 1. Your PR will be reviewed by the maintainers
 2. If everything looks good, one of them will `approve` it, and your PR will be merged.

Thank you for contributing!

eeyun commented:
fixes https://github.com/habitat-sh/habitat/issues/2862

eeyun commented:
@thesentinels acceptance

thesentinels commented:
:metal: I am sending your branch against master to our acceptance environment.

thesentinels commented:
:neckbeard: Travis CI has [started the deployment of acceptance](https://travis-ci.org/habitat-sh/habitat/builds/262837301).

thesentinels commented:
:neckbeard: Travis CI has [started the deployment of acceptance](https://travis-ci.org/habitat-sh/habitat/builds/262837301).

thesentinels commented:
:sparkling_heart: Travis CI [reports acceptance has been deployed](https://travis-ci.org/habitat-sh/habitat/builds/262837301).

I love new and shiny websites!

I just want you and the contributor to answer me one question:

![gif-keyboard-3280869874741411265](https://cloud.githubusercontent.com/assets/4304/17755674/1e577b82-6490-11e6-96b7-1663d283c824.gif)

mgamini commented on www/source/stylesheets/_global.scss:
```diff
@@ -171,7 +171,7 @@ body {
 // ---------
 pre {
-  background-color: $hab-navy;
+  background-color: #2B303B;
```
This is pretty close to `$hab-navy-2` - does that not look ok?

eeyun commented:
It's close. The current setting is a little darker. If we want to set this
back to the bluer color we can modify the ocean css to match the background
color here!

On Wed, Aug 9, 2017, 6:28 PM Garrett Amini <notifications@github.com> wrote:

> *@mgamini* approved this pull request.
>
> ------------------------------
>
> In www/source/stylesheets/_global.scss
> <https://github.com/habitat-sh/habitat/pull/2901#discussion_r132320485>:
>
> > @@ -171,7 +171,7 @@ body {
>  // 2. Code Overrides
>  // ---------
>  pre {
> -  background-color: $hab-navy;
> +  background-color: #2B303B;
>
> This is pretty close to $hab-navy-2 - does that not look ok?
>
> —
> You are receiving this because you were assigned.
> Reply to this email directly, view it on GitHub
> <https://github.com/habitat-sh/habitat/pull/2901#pullrequestreview-55388756>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AMB97HA8wuc6-y8iZ5Zamsq-Dt2Q20dAks5sWjKdgaJpZM4Oyoe5>
> .
>

apriofrost commented:
Thanks for the awesome work, Ian! I have a few feedback on the colors:
1. One thing we might need to pay attention to is the accessibility of the colors, especially for the single line command under the Interactive Demo section (http://www.acceptance.habitat.sh/tutorials/get-started/demo/). Because currently, a user has to read and type in the command manually (we hasn't implemented the copy pasting feature yet). It has to be very easy to read. The color contrast shown in the screenshot below doesn't satisfy that (I haven't tried with a color contrast checker, but it's unlikely to pass). I would suggest that we just use the current color on the production site.
![image](https://user-images.githubusercontent.com/9401459/29147339-3a4ce3f0-7d1b-11e7-97d1-46ab0962d733.png)
2. The background color of the simulated terminal doesn't match with the new background color of the command line. It's better to keep them the same color when they appear close together.  I would prefer to keep it as the current dark blue color (#283c4c) since it matches better with the current visual style across the site.
![image](https://user-images.githubusercontent.com/9401459/29147605-f1960c84-7d1c-11e7-83f9-7aed8edb9baf.png)

eeyun commented:
Ah yep this is due to incorrect syntax on the command. If we update the
content to start with a $ then it should autocorrect to appropriate
highlighting

On Wed, Aug 9, 2017, 7:08 PM Liz Chen <notifications@github.com> wrote:

> Thanks for the awesome work, Ian! I have a few feedback on the colors:
>
>    1. One thing we might need to pay attention to is the accessibility of
>    the colors, especially for the single line command under the Interactive
>    Demo section (
>    http://www.acceptance.habitat.sh/tutorials/get-started/demo/). Because
>    currently, a user has to read and type in the command manually (we hasn't
>    implemented the copy pasting feature yet). It has to be very easy to read.
>    The color contrast shown in the screenshot below doesn't satisfy that (I
>    haven't tried with a color contrast checker, but it's unlikely to pass). I
>    would suggest that we just use the current color on the production site.
>    [image: image]
>    <https://user-images.githubusercontent.com/9401459/29147339-3a4ce3f0-7d1b-11e7-97d1-46ab0962d733.png>
>    2. The background color of the simulated terminal doesn't match with
>    the new background color of the command line. It's better to keep them the
>    same color when they appear close together. I would prefer to keep it as
>    the current dark blue color (#283c4c) since it matches better with the
>    current visual style across the site.
>    [image: image]
>    <https://user-images.githubusercontent.com/9401459/29147605-f1960c84-7d1c-11e7-83f9-7aed8edb9baf.png>
>
> —
> You are receiving this because you were assigned.
> Reply to this email directly, view it on GitHub
> <https://github.com/habitat-sh/habitat/pull/2901#issuecomment-321405866>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AMB97MXQpP0c5qfx1V6i5OSDExjdx9azks5sWjvmgaJpZM4Oyoe5>
> .
>

apriofrost commented:
Cool! I chatted with @davidwrede about this and he currently doesn't have the bandwidth to update the content to include the $ sign. @eeyun Do you think it will be a good idea to suspend on the syntax highlighting for now until we have a better understanding of what color palette works best? This is a great thing to do to improve the readability for our users, just that accessibility and branding need also be taken into consideration.

eeyun commented:
I can get these pages updated as well. It should be trivial to roll that into this PR. I also don't think we should hold off on syntax highlighting. The code is in and functional and doesn't break anything. I don't mind tweaking the initial colors to get it close to begin with. Afterwards we can come back and iterate on it if we want some really granular and specific palette for the highlights themselves, but I don't think there's a really strong reason to not get the update in and worry about that later. With the exception of the background the highlight colors are very minimal so it shouldn't be an issue.

apriofrost commented:
Cool! As long as we can replace the gray color appeared in the '--peer' in the screenshot with a color that contrasts well with the background, I'm cool with it. 
If you can make the adjustment to the initial colors, here are some to consider (it's the habitat colors that contrast relatively well with the dark blue background):
#FF9012
#EB6852
#87B09A
#C3C6C8

Thanks so much Ian!

eeyun commented:
Ok. I've corrected all of the syntax highlighting colors to match the habitat brand colors from brand.chef.io which are the same colors you've posted above @apriofrost . I've also corrected an issue with the new markdown engine that wasn't rendering tables. And I've added correct syntax to the demo pages so the syntax highlighting should be correct there. I'm going to go ahead and push this out to acceptance @apriofrost @cnunciato @mgamini feel free to check it out www.acceptance.habitat.sh would like to get this merged today!

eeyun commented:
One other note: For the background color of the codeblocks I've switched it to match the deep greenish blue of the demo terminal.

eeyun commented:
@thesentinels acceptance

thesentinels commented:
:metal: I am sending your branch against master to our acceptance environment.

thesentinels commented:
:neckbeard: Travis CI has [started the deployment of acceptance](https://travis-ci.org/habitat-sh/habitat/builds/263106349).

thesentinels commented:
:sparkling_heart: Travis CI [reports acceptance has been deployed](https://travis-ci.org/habitat-sh/habitat/builds/263106349).

I love new and shiny websites!

I just want you and the contributor to answer me one question:

![gif-keyboard-3280869874741411265](https://cloud.githubusercontent.com/assets/4304/17755674/1e577b82-6490-11e6-96b7-1663d283c824.gif)

eeyun commented:
@thesentinels approve

thesentinels commented:
:metal: I am testing your branch against master before merging it. We do this to ensure that the master branch is never failing tests.

thesentinels commented:
:neckbeard: Travis CI has [started testing this PR](https://travis-ci.org/habitat-sh/habitat/builds/263176934).

thesentinels commented:
:sparkling_heart: Travis CI [reports this PR passed](https://travis-ci.org/habitat-sh/habitat/builds/263176934).

It always makes me feel nice when humans approve of one anothers work. I'm merging this PR now.

I just want you and the contributor to answer me one question:

![gif-keyboard-3280869874741411265](https://cloud.githubusercontent.com/assets/4304/17755674/1e577b82-6490-11e6-96b7-1663d283c824.gif)
